engine/src/main/java/org/opencds/cqf/cql/engine/elm/execution/ToQuantityEvaluator.java

```diff
@@ -48,28 +48,26 @@
 public class ToQuantityEvaluator extends org.cqframework.cql.elm.execution.ToQuantity {
 
     public static Quantity toQuantity(String str) {
-        StringBuilder number = new StringBuilder();
-        StringBuilder unit = new StringBuilder();
-        for (char c : str.toCharArray()) {
-            if ((Character.isDigit(c) || c == '.' || c == '+' || c == '-') && unit.length() == 0) {
-                number.append(c);
-            } else if (Character.isLetter(c) || c == '/') {
-                unit.append(c);
-            } else if (c == ' ' || c == '\'') {
-                // continue
-            } else {
-                throw new IllegalArgumentException(String.format("%c is not allowed in ToQuantity format", c));
-            }
-        }
-        try {
-            BigDecimal ret = new BigDecimal(number.toString());
-            if (Value.validateDecimal(ret, null) == null) {
+        String[] tokens = str.trim().replaceAll("\'", "").split("\\s+");
```

Comment:
Here's the definition of what UCUM units look like (section 2.1):
https://ucum.org/ucum.html

And here's some sample UCUM units for reference:

https://ucum.org/trac/attachment/wiki/adoption/common/TableOfExampleUcumCodesForElectronicMessagingWithPreface.pdf

I'd like to see tests for a few of those to make sure they parse correctly since they can be quite complex. White space within the unit should either be ignored or an error thrown. I don't think it's generally safe to just split on the white space since that'll give you partial units if there's bad input.